src/DataSet/ObjectDataSet.php

```diff
@@ -54,64 +129,52 @@ public function getAttributeValue(string $attribute): mixed
             return $this->object->getAttributeValue($attribute);
         }
 
-        return $this->hasAttribute($attribute)
-            ? $this->reflectionProperties[$attribute]->getValue($this->object)
-            : null;
+        return $this->getData()[$attribute] ?? null;
     }
 
     public function hasAttribute(string $attribute): bool
     {
         return $this->dataSetProvided
             ? $this->object->hasAttribute($attribute)
-            : array_key_exists($attribute, $this->reflectionProperties);
+            : array_key_exists($attribute, $this->getReflectionProperties());
     }
 
     public function getData(): array
     {
-        if ($this->dataSetProvided) {
-            return $this->object->getData();
+        if ($this->data !== null) {
```

Comment:
If object provided `DataSetInterface` then not need cache and need use `getData()` from object.

```php
if ($this->dataSetProvided) {
        return $this->object->getData();
}
````